proto/internal/temporal/server/api/persistence/v1/queues.proto

```diff
@@ -61,3 +61,20 @@ message HistoryTask {
     // switch on the task category, which should be available from the metadata for the queue that this task came from.
     temporal.api.common.v1.DataBlob blob = 2;
 }
+
+
+message QueuePartition {
+  // min_message_id is less than or equal to the id of every message in the queue. If there are no messages in the
+  // queue, then this is 0. It is also 0 if there has only been one message enqueued and it has not been deleted yet
+  // because the first message is always assigned id 0. The min_message_id is mainly used to skip over tombstones in
+  // Cassandra: let's say we deleted the first 1K messages from a queue with 1.1K messages. If an operator asked for
+  // the first 100 messages, without the min_message_id, we would have to scan over the 1K tombstone rows before we
+  // could return the 100 messages. With the min_message_id, we can skip over all of the tombstones by specifying
+  // message_id >= queue.min_message_id.
+  int64 min_message_id = 1;
```

Comment:
same comment as in the other PR re. ranges ideally should be half open. There's no correctness or performance issue though, so I'll leave it to you to decide.

Reply:
Hmm, how would it be half-open in this case?

Reply:
Discussed offline: we are going to keep this as-is